MaxText/layers/linears.py

```diff
@@ -623,7 +645,9 @@ def maybe_all_gather_kernel_weight_in_expert_parallelism(self, kernel, kernel_ax
 
   def dense_matmul(self, inputs, gate_logits, w0_kernel, w1_kernel, wo_kernel):
     # gate_logits: batch, length, expert
+    # follow router_logits = shd.shard(router_logits, (None, None, None))
```

Comment:
Nit: what's the `shd.shard`? or you mean `nn.with_logical_constraint`?

Reply:
that's old comments from Zhiyu, cleaned it up now